crates/turbo-tasks/src/graph/control_flow.rs

```diff
@@ -1,7 +1,9 @@
+use tracing::Span;
+
 /// The control flow of visiting an edge during a graph traversal.
 pub enum VisitControlFlow<Node, Abort = !> {
     /// The traversal should continue on the outgoing edges of the given node.
-    Continue(Node),
+    Continue(Node, Span),
```

Comment:
Can we have the knowledge of `Span` live at the visitor level instead? Where `Span` would be a property of the `Node`, and the visitor would require some `Node: WithSpan` bound.

Something like the following:

```
ReverseTopological::new().skip_duplicates().visit(
    [node],
    |node| {
        async move { ... }
    }.instrument()
)
```